sympy/core/basic.py

```diff
@@ -1525,6 +1525,52 @@ def match(self, pattern, old=False):
         pattern = sympify(pattern)
         return pattern.matches(self, old=old)
 
+    def partition(self, var):
```

Comment:
Maybe we should make it private, e.g. by calling it `_partition`. Maybe `_partition_args` could be a more suitable name. Is it intended for end-users?